parlai/crowdsourcing/tasks/turn_annotations_static/webapp/src/components/task_components.jsx

```diff
@@ -184,125 +49,352 @@ function SubtaskCounter({ subtaskIndex, numSubtasks }) {
-        <span><b>{turnIdx % 2 == 0 ? 'YOU' : 'THEM'}:</b> {text}
-          <ErrorBoundary>
-            {extraElements}
-          </ErrorBoundary>
+
+  setLastSelected = (checkbox) => {
+    this.setState({lastSelected: checkbox});
+  }
+
+  setLastSubtaskIdx = (subtaskIdx) => {
+    this.setState({lastSubtaskIdx: subtaskIdx});
+  }
+
+  handleResponseFieldChange = evt => {
+    this.props.setResponse(this.props.turnIdx, evt.target.value);
+  };
+
+  render() {
+    const {
+      annotationQuestion,
+      annotationBuckets,
+      turnIdx,
+      turnAnswers,
+      doAnnotateMessage,
+      askReason,
+      askResponse,
+      setChecked,
+      setReason,
+    } = this.props;
+    const {
+      lastSelected,
+    } = this.state;
+    var extraElements = '';
+    var responseInputElement = '';
+    if (doAnnotateMessage) {
+      if (annotationBuckets !== null) { 
+        extraElements = (
+          <span key={'extra_' + turnIdx}>
+            <br />
+            <br />
+            <span style={{fontStyle: 'italic'}}>
+              {annotationQuestion}
+              <br />
+              <Checkboxes
+                turnIdx={turnIdx}
+                turnAnswers={turnAnswers}
+                annotationBuckets={annotationBuckets}
+                askReason={askReason}
+                lastSelected={lastSelected}
+                setLastSelected={this.setLastSelected}
+                setChecked={setChecked}
+                setReason={setReason}
+              />
+            </span>
+          </span>
+        );
+      }
+      if (askResponse !== null) {
+        responseInputElement = (
+          <FormControl
+          type="text"
+          name="input_response"
+          id={"input_response_" + turnIdx}
+          style={{
+              fontSize: "14px",
+              resize: "none",
+              marginBottom: "40px"
+          }}
+          onChange={this.handleResponseFieldChange}
+          value={turnAnswers.response}
+          placeholder={"Please enter your response here"}
+          onPaste={(e) => {e.preventDefault(); alert("Please do not copy and paste. You must manually respond to each message.")}}
+          autoComplete="off"
+        />
+        )
+      }
+    }
+    const agentIdx = turnAnswers.agent_idx;
+    return (
+      <div
+        className={`alert ${agentIdx == 0 ? 'alert-info' : 'alert-warning'}`}
+        style={{
+          float: `${agentIdx == 0 ? 'right' : 'left'}`,
+          display: 'table',
+          minWidth: `${agentIdx == 0 ? '30%' : '80%'}`,
+          marginTop: `${turnIdx == 1 ? '40px' : 'auto'}`,
+        }}>
+        <span>
+          <b>{turnIdx % 2 == 0 ? 'YOU' : 'THEM'}:</b> {turnAnswers.text}
+          <ErrorBoundary>{extraElements}</ErrorBoundary>
         </span>
       </div>
-      {responseInputElement}
```

Comment:
this line / the divs around it needs to be ported over so the `responseInputElement` aka the freeform textbox will show up!